### ISSUE TITLE
IDE0055 false positive with `[..XXX]` pattern

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn#73742]
**Version Used**: MicrosoftNetCompilersTollsetVersion = 4.6.0

**Steps to Reproduce**: Look at PR https://github.com/microsoft/testfx/pull/3009

**Diagnostic Id**: IDE0055

**Expected Behavior**: No diagnostic

**Actual Behavior**: Diagnostic

![image](https://github.com/dotnet/roslyn/assets/11340282/692e4e16-6fd4-4df3-89c3-d65e7ae143cc)

Sample code:

```cs
Console.WriteLine("Hello, World!");

var executableInfo = new ExecutableInfo("dotnet", new[] { "run" }, "C:/workspace");

List<string> list2 = [..executableInfo.Arguments]; // IDE0055

Console.WriteLine(list2);

internal sealed class ExecutableInfo(string fileName, IEnumerable<string> arguments, string workspace)
{
    public string FileName { get; } = fileName;

    public IEnumerable<string> Arguments { get; } = arguments;

    public string Workspace { get; } = workspace;
}
```

**Comment by CyrusNajmabadi** [comment]
I don't have access to that repro.  Can you please crete a small sample and update your OP with it?

**Comment by Evangelink** [comment]
I am surprised you cannot access the repo as it's OSS but anyways editing the OP with a sample.

**Comment by CyrusNajmabadi** [comment]
> I am surprised you cannot access the repo as it's OSS but anyways editing the OP with a sample.

I cannot access Microsoft org resources as they require logging in and validating machines, and that doesn't work on non-domain-joined machines from MS accounts.

**Comment by CyrusNajmabadi** [comment]
Closing out as by design.  Your formatting isn't following the requirements we have for this feature.  You need to write it as `[.. foo]` as the space is mandatory.  THis matches the same rule we have for `..` in a pattern.  e.g. `x is ['a', .. var mid, 'z']`.

**Comment by SimonCropp** [comment]
i would argue it is still broken, as it tells u to fix formatting without telling u what it wants fixed

**Comment by CyrusNajmabadi** [comment]
> i would argue it is still broken, as it tells u to fix formatting without telling u what it wants fixed

Runnign the fixer will add the space back in.

**Comment by SimonCropp** [comment]
@CyrusNajmabadi that doesnt fix the problem. in the case we are talking about it failed in CI. any build failure should unambiguously explain what the problem is

**Comment by CyrusNajmabadi** [comment]
IDE0055 has always stated that hte problem is that the formatting is incorrect at that particular location.  The fix is to fix the formatting.

**Comment by SimonCropp** [comment]
yes. but what is incorrect about the formatting is valuable.

**Comment by SimonCropp** [comment]
> IDE0055 has always stated that hte problem is that the formatting is incorrect at that particular location

to me that just reads as "IDE0055 has always been subtly broken"

**Comment by CyrusNajmabadi** [comment]
> yes. but what is incorrect about the formatting is valuable.

Formatting changes whitespace.  So what is broken here is the whitespace you have.  That's it.  All we could say is "incorrect whitespace".  Which is the same as "your formatting is incorrect".

**Comment by CyrusNajmabadi** [comment]
> to me that just reads as "IDE0055 has always been subtly broken"

ALl our features do this.  They tell you you have a problem.  What the fix may be is feature dependent, and even depends on what fix you choose in that case (a specific problem may have a wealth of different fixes available).  How you fix it is up to you.

**Comment by SimonCropp** [comment]
> All we could say is "incorrect whitespace". Which is the same as "your formatting is incorrect".

"formatting" is ambiguous. "whitespace" is less so

**Comment by SimonCropp** [comment]
> All our features do this. They tell you you have a problem. 

AKA "All our features do this. They tell you you have a problem, not what the problem is"?
